docs/architecture/adr-040-storage-and-smt-state-commitments.md

```diff
@@ -83,14 +85,14 @@ NOTE: For the SDK storage, we may consider splitting that interface into `Commit
 
 Number of historical versions (snapshots) for `abci.Query` and fast sync is part of a node configuration, not a chain configuration. A configuration should allow to specify number of past blocks and number of past blocks modulo some number (eg: 100 past blocks and one snapshot every 100 blocks for past 2000 blocks). Archival nodes can keep all snapshots.
 
-Pruning old snapshots is effectively done by DB. Whenever we update a record in SC, SMT will create a new one without removing the old one. Since we are using a snapshot for each block, we must update the mechanism and immediately remove an orphaned from the storage. This is a safe operation - snapshots will keep track of the records which should be available for past versions.
+Pruning old snapshots is effectively done by DB. Whenever we update a record in `SC`, SMT will create a new one without removing the old one. Since we are using a snapshot for each block, we must update the mechanism and immediately remove an orphaned from the storage. This is a safe operation - snapshots will keep track of the records which should be available for past versions.
 
 To manage the active snapshots we will either us a DB _max number of snapshots_ option (if available), or will remove snapshots in the `EndBlocker`. The latter option can be done efficiently by identifying snapshots with block height.
 
 #### Accessing old state versions
 
 One of the functional requirements is to access old state. This is done through `abci.Query` structure.  The version is specified by a block height (so we query for an object by a key `K` at block height `H`). The number of old versions supported for `abci.Query` is configurable. Accessing an old state is done by using available snapshots.
-`abci.Query` doesn't need old state of SC. So, for efficiency, we should keep SC and SM Storage in different databases (however using the same DB engine). We will only create snapshots for
+`abci.Query` doesn't need old state of `SC`. So, for efficiency, we should keep `SC` and `SS` in different databases (however using the same DB engine). We will only create snapshots for
```

Comment:
Multiple database instances will compete for resources (CPU, memory, IO). IMHO without benchmarking, you can't say that this will be beneficial for efficiency.

Reply:
My thinking was that `SC` doesn't need to be part of the transactions - we don't need to keep old versions, as we need with  `SS` (State Store). If we don't separate this DB, then it will be hard to keep versioning for `SS` and not do it for `SC`. Does it make sense? If yes I will update the paragraph above.